src/github/api/operations.rs

```diff
@@ -0,0 +1,93 @@
+use octocrab::models::issues::Comment;
+use reqwest::StatusCode;
+
+use crate::github::api::RepositoryClient;
+
+#[derive(Debug)]
+pub enum MergeError {
+    NotFound,
+    Conflict,
+    AlreadyMerged,
```

Comment:
Probably would be a good idea to hold on to the refs that could not be merged in this errors.

Reply:
The caller of the function passes the head/base refs, so I don't think that it's strictly needed here. I'll add them if there will be a callsite where it will be needed.